mk/tools.mk

```diff
@@ -318,3 +318,10 @@ breakpad_clean:
 	$(V1) [ ! -d "$(BREAKPAD_DIR)" ] || $(RM) -rf $(BREAKPAD_DIR)
 	@echo " CLEAN        $(BREAKPAD_DL_FILE)"
 	$(V1) $(RM) -f $(BREAKPAD_DL_FILE)
+
+.PHONY: lint
+lint:
+	@command -v git >/dev/null 2>&1 || { echo >&2 "Error: git is required for linting, but it's not installed. Aborting."; exit 1; }
+	@command -v clang-format >/dev/null 2>&1 || { echo >&2 "Error: clang-format is required for linting, but it's not installed. Aborting."; exit 1; }
+	@command -v git-clang-format >/dev/null 2>&1 || { echo >&2 "Error: git-clang-format is required for linting, but it's not installed. Aborting."; exit 1; }
+	git clang-format --style=file --extensions=c,h --verbose -- src
```

Comment:
```suggestion
	git clang-format --style=file --extensions=c,h --verbose --src
```

Reply:
It's not a flag or an option, rather an extra positional argument I was following the usage of `git-clang-format`, by running:

```bash
git-clang-format --help
```

`--src` flag doesn't seems to exist.

That's the usage:
```
usage: git clang-format [OPTIONS] [<commit>] [<commit>|--staged] [--] [<file>...]

If zero or one commits are given, run clang-format on all lines that differ
between the working directory and <commit>, which defaults to HEAD.  Changes are
only applied to the working directory, or in the stage/index.

If two commits are given (requires --diff), run clang-format on all lines in the
second <commit> that differ from the first <commit>.

The following git-config settings set the default of the corresponding option:
  clangFormat.binary
  clangFormat.commit
  clangFormat.extensions
  clangFormat.style

positional arguments:
  <commit>              revision from which to compute the diff
  <file>...             if specified, only consider differences in these files

options:
  -h, --help            show this help message and exit
  --binary BINARY       path to clang-format
  --commit COMMIT       default commit to use if none is specified
  --diff                print a diff instead of applying the changes
  --diffstat            print a diffstat instead of applying the changes
  --extensions EXTENSIONS
                        comma-separated list of file extensions to format, excluding the period and case-insensitive
  -f, --force           allow changes to unstaged files
  -p, --patch           select hunks interactively
  -q, --quiet           print less information
  --staged, --cached    format lines in the stage instead of the working dir
  --style STYLE         passed to clang-format
  -v, --verbose         print extra information
```

Paths or file names would be provided at the end of the command following the `--` which indicates the end of the git-clang-format binary command options / flags. I think those are the positional argument that are going directly to clang-format binary, they probably didn't want to duplicate functionality in this plugin so rather they just pass it through.
So similarly: `clang-format <filename>`, I guess that's what the plugin is doing internally.

Reply:
https://github.com/llvm/llvm-project/blob/7048857f52005810a2d8f43e52e659bfd6342521/clang/tools/clang-format/git-clang-format#L260-L263